arm-datalake-analytics/job/2016-11-01/swagger/job.json

```diff
@@ -0,0 +1,1070 @@
+            "name": "SeverityTypes",
+            "modelAsString": false
+          }
+        },
+        "source": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the ultimate source of the failure (usually either SYSTEM or USER)."
+        },
+        "startOffset": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int32",
+          "description": "the start offset in the job where the error was found"
+        }
+      },
+      "description": "The Data Lake Analytics job error details."
+    },
+    "JobInnerError": {
+      "properties": {
+        "diagnosticCode": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int32",
+          "description": "the diagnostic error code."
+        },
+        "severity": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the severity level of the failure.",
+          "enum": [
+            "Warning",
+            "Error",
+            "Info"
+          ],
+          "x-ms-enum": {
+            "name": "SeverityTypes",
+            "modelAsString": false
+          }
+        },
+        "details": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the details of the error message."
+        },
+        "component": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the component that failed."
+        },
+        "errorId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the specific identifier for the type of error encountered in the job."
+        },
+        "helpLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the link to MSDN or Azure help for this type of error, if any."
+        },
+        "internalDiagnostics": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the internal diagnostic stack trace if the user requesting the job error details has sufficient permissions it will be retrieved, otherwise it will be empty."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the user friendly error message for the failure."
+        },
+        "resolution": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the recommended resolution for the failure, if any."
+        },
+        "source": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the ultimate source of the failure (usually either SYSTEM or USER)."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the error message description"
+        }
+      },
+      "description": "The Data Lake Analytics job error details."
+    },
+    "JobInformation": {
+      "properties": {
+        "jobId": {
```

Comment:
I agree with your assessment. I will make this read-only to be explicit, since it is ignored if passed in the payload. This is a breaking change, though.